upgrade-2.x.sh

```diff
@@ -0,0 +1,201 @@
+#!/bin/bash
```

Comment:
I posted a rationale for this once. Imagine you have the file open with a text editor. You then cannot execute it by running ./a.sh. So you run sh a.sh. Which then, perhaps only after executing half of the script, crashes with a syntax error.